sherpa/conftest.py

```diff
@@ -64,7 +64,14 @@ def pytest_addoption(parser):
             r"File '/data/regression_test/master/in/sherpa/aref_Cedge.fits' does not have write permission.  Changing to read-only mode."
         ],
     RuntimeWarning:
-        [r"invalid value encountered in sqrt", ],
+        [r"invalid value encountered in sqrt",
+         # The following is added for the test_can_use_swift_data
+         # test in sherpa/astro/tests/test_astro_data_swift_unit.py;
+         # as the test checks this warning is created it would be useful
+         # if the test could eitehr clear the warnings, or tell this
+         # fixture to ignore it, rather than making it a global flag
+         r"divide by zero encountered in divide",
```

Comment:
I managed to forget about this comment until this morning. I haven't looked at it yet (unlikely to get around to it until tonight now). I don't know if it's related that in pytest 3 they added support for reporting warnings, so I don't know if any of the things we are doing interact strangely with that code (I doubt it but just mention it). My experience with the warning code has been some fragility, in that some things don't seem reliable. This is an aside, and we should work out what's going on here before merging the code.